### PR TITLE
PRI domain support

### PR DESCRIPTION
**Pull request by farski**
This (hopefully) adds support to handle traffic to `pri.org` and `www.pri.org`, and redirect **all** of it to either `theworld.org` or a subdomain of theworld.org.

Also allows for redirects to be 301s, rather than 302s.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -22,33 +32,52 @@ const ROUTES = [
...
+
+  if (PRI_HOSTS.includes(headers.host)) {
+    // Handle pri.org traffic
+    PRI_ROUTES.find(([matchers, obj]) => {
+      if (matchers.some(m => m.test(event.path))) {
+        return route = obj;
+      }
     });
-  }
+  } else {
```
Comment (cavis) [inline index.js:57]: Might be cleaner to now split this into 3 conditions ...

- `PRI_HOSTS.includes(headers.host)`
- `HOSTS.includes(headers.host)` (probably safe to lose the HOSTS.length check)
- `else` ... 302 to the PRX canonical host

File: lib/redirect.js [hunk 2]
```diff
@@ -7,10 +7,11 @@ const util = require('./util');
  */
 module.exports = class Redirect {
 
-  constructor(host, rewrite, useHttps = true) {
+  constructor(host, rewrite, useHttps = true, statusCode = 302) {
     this.host = host;
     this.rewrite = rewrite;
     this.protocol = useHttps ? 'https' : 'http';
+    this.statusCode = statusCode;
```
Comment (cavis) [inline lib/redirect.js:14]: Test for this?

File: index.js [hunk 3]
```diff
@@ -22,33 +32,52 @@ const ROUTES = [
...
-  // make sure we're at a canonical host
-  if (HOSTS.length && headers['host'] && HOSTS.indexOf(headers['host']) === -1) {
-    const loc = `https://${HOSTS[0]}${event.path}${util.queryToString(event.queryStringParameters)}`;
-    return callback(null, {
-      statusCode: 302,
-      headers: {'location': loc, 'content-type': 'text/plain', },
-      body: `Moved to ${loc}`
+  let route;
+
+  if (PRI_HOSTS.includes(headers.host)) {
```
Comment (cavis) [inline index.js:50]: I kind of wish there were some tests against the handler now ... as the amount of logic going on here increases.  But probably out-of-scope for this PR.

File: routes/pri-redirect.js [hunk 4]
```diff
@@ -1,6 +0,0 @@
-'use strict';
-
-/**
- * Paths redirected to www.pri.org
- */
-module.exports = [];
```
Comment (cavis) [inline routes/pri-redirect.js:6]: What was this file?  I still see a require for it in `index.js` too ... the `npm start` dev server fails to start.

Comment (cavis) [reply routes/pri-redirect.js:6]: Also an unused `const PRI_HOST` in there.

File: index.js (hunk 3 above)
Comment (cavis) [reply index.js:50]: FWIW, you can spot check these locally with `npm start` ...

```
curl -i localhost:3000/some/route -H 'Host: pri.org'
```

**Review by cavis — CHANGES_REQUESTED** [review]
Couple leftover pieces of code.  And wishing I'd tested the index file.  But seems like this will work fine to also proxy pri.org stuff.

File: routes/pri-redirect.js (hunk 4 above)
Comment (farski) [reply routes/pri-redirect.js:6]: It was for `/support`, which res dev asked to be removed a few weeks/months back. It has a real page on Squarespace now

**Review by cavis — APPROVED** [review]
👍 looks good to me!  May want to QA in staging a bit, but can't imagine it'll break PRX domains.

**Pull request merged**